lib/stytch/oauth.rb

```diff
@@ -1,54 +1,161 @@
+    #   The type of this field is nilable `String`.
+    #
+    # == Returns:
+    # An object with the following fields:
+    # request_id::
+    #   Globally unique UUID that is returned with every API call. This value is important to log for debugging purposes; we may ask for this value to help identify a specific API call when helping you debug an issue.
+    #   The type of this field is `String`.
+    # oauth_attach_token::
+    #   A single-use token for connecting the Stytch User selection from an OAuth Attach request to the corresponding OAuth Start request.
+    #   The type of this field is `String`.
+    # status_code::
+    #   The HTTP status code of the response. Stytch follows standard HTTP response status code patterns, e.g. 2XX values equate to success, 3XX values are redirects, 4XX are client errors, and 5XX are server errors.
+    #   The type of this field is `Fixnum`.
+    def attach(
+      provider:,
+      user_id: nil,
       session_token: nil,
-      session_jwt: nil,
-      session_duration_minutes: nil,
-      session_custom_claims: nil,
-      code_verifier: nil
+      session_jwt: nil
     )
       request = {
-        token: token
+        provider: provider
       }
+      request[:user_id] = user_id unless user_id.nil?
       request[:session_token] = session_token unless session_token.nil?
       request[:session_jwt] = session_jwt unless session_jwt.nil?
-      request[:session_duration_minutes] = session_duration_minutes unless session_duration_minutes.nil?
-      request[:session_custom_claims] = session_custom_claims unless session_custom_claims.nil?
-      request[:code_verifier] = code_verifier unless code_verifier.nil?
 
-      post_request("#{PATH}/authenticate", request)
+      post_request('/v1/oauth/attach', request)
     end
 
-    # Send a /v1/oauth/attach request.
+    # Authenticate a User given a `token`. This endpoint verifies that the user completed the OAuth flow by verifying that the token is valid and hasn't expired. To initiate a Stytch session for the user while authenticating their OAuth token, include `session_duration_minutes`; a session with the identity provider, e.g. Google or Facebook, will always be initiated upon successful authentication.
     #
-    # Exactly one of user_id, session_token, or session_jwt is required.
-    def attach(
-      provider:,
-      user_id: nil,
+    # == Parameters:
+    # token::
+    #   The token to authenticate.
+    #   The type of this field is `String`.
+    # session_token::
+    #   (Stytch sessions only) Reuse an existing session instead of creating a new one. If you provide us with a `session_token`, then we'll update the session represented by this session token with this OAuth factor. If this `session_token` belongs to a different user than the OAuth token, the session_jwt will be ignored. This endpoint will error if both `session_token` and `session_jwt` are provided.
```

Comment:
sort of weird how long this line is, when the next description has line breaks in it. Not sure if that actually matters to the editor displaying the docs, though